app/src/@libs/evm-wallet/providers/EvmWalletProvider.tsx

```diff
@@ -34,7 +36,7 @@ export function EvmWalletProvider({ children }: EvmWalletProviderProps) {
   const address = data ? data.address : undefined;
   const chainId = data ? data.chainId : undefined;
   const error = data ? data.error : undefined;
-  const provider = data ? data.provider : undefined;
+  const provider = data ? (data.provider as Web3Provider) : undefined;
```

Comment:
Now the type is `BaseProvider | Web3Provider | undefined.` Is casting to `Web3Provider` is safe here?